simtools/corsika/corsika_output.py

```diff
@@ -0,0 +1,1002 @@
+        """
+        Extract the information of the Cherenkov photons from a CORSIKA output IACT file, create
+         and fill the histograms
+
+        Parameters
+        ----------
+        telescope_indices: int or list of int
+            The indices of the specific telescopes to be inspected. If not specified, all telescopes
+            are treated together in one histogram.
+
+        Returns
+        -------
+        list: list of boost_histogram.Histogram instances.
+
+        """
+        self._telescope_indices = telescope_indices
+        self._create_histograms()
+
+        self.num_photons_per_event_per_telescope = []
+        start_time = time.time()
+        self._logger.debug("Starting reading the file at {}.".format(start_time))
+        with IACTFile(self.input_file) as f:
+            event_counter = 0
+            for event in f:
+                for step, _ in enumerate(self._tel_positions):
+                    self.num_photons_per_event_per_telescope.append(event.n_photons[step])
+
+                photons = list(event.photon_bunches.values())
+                self._fill_histograms(
+                    photons,
+                    self.event_azimuth_angles[event_counter],
+                    self.event_zenith_angles[event_counter],
+                )
+                event_counter += 1
+        self._logger.debug(
+            "Finished reading the file and creating the histograms in {} seconds".format(
+                time.time() - start_time
+            )
+        )
+
+    def _raise_if_no_histogram(self):
+        """
+        Raise an error if the histograms were not created.
+
+        Raises
+        ------
+        HistogramNotCreated:
+            if the histogram was not previously created.
+        """
+
+        for histogram in self._allowed_histograms:
+            if not hasattr(self, histogram):
+                msg = (
+                    "The histograms were not created. Please, use `create_histograms` to create "
+                    "histograms from the CORSIKA output file."
+                )
+                self._logger.error(msg)
+                raise HistogramNotCreated
+
+    def _get_2D(self, label):
+        """
+        Helper function to get 2D distributions.
+
+        Parameters
+        ----------
+        label: str
+            Label to indicate which histogram.
+
+        Raises
+        ------
+        ValueError:
+            if label is not valid.
+        """
+
+        if label not in self._allowed_2D_labels:
+            msg = "label is not valid. Valid entries are {}".format(self._allowed_2D_labels)
+            self._logger.error(msg)
+            raise ValueError
+        self._raise_if_no_histogram()
+
+        if self.telescope_indices is None:
+            size = 1
+        else:
+            size = len(self.telescope_indices)
+
+        x_edges, y_edges, hist_values = [], [], []
+        for step in range(size):
```

Comment:
I did change the names, but I preferred using `one_telescope` instead of `i_telescope` and `num_telescopes` instead of `n_telescopes`. do you have a strong opinion on that?

Reply:
I don't mind `num_telescope`, but `one_telescope` is not correct cause it's an iterator and iterators should be `i_x`.